source/dem/pw_contact_force.cc

```diff
@@ -98,5 +100,278 @@ PWContactForce<dim>::update_contact_information(
   contact_info.tangential_relative_velocity = tangential_relative_velocity;
 }
 
+template <int dim>
+Tensor<1, dim>
+  PWContactForce<dim>::calculation_total_torque(Tensor<1, dim> total_force,
+                                                Point<dim>     center_mass,
+                                                Point<dim> point_on_boundary)
+{
+  if (dim == 3)
+    {
+      return cross_product_3d(total_force, point_on_boundary - center_mass);
+    }
+}
+
+template <int dim>
+void
+PWContactForce<dim>::write_forces_torques(std::map<unsigned int, Tensor<1, dim>> total_force,
+                     std::map<unsigned int, Tensor<1, dim>> total_torque,
+                     std::basic_string<char> name_file,
+                     double current_time)
+{
+  std::ofstream myFile;
+  myFile.open(name_file,std::ios::app);
+  myFile    << "Time "<<current_time<<" seconds\n";
+  for (const auto& it : total_force)
+    {
+      myFile    <<"Boundary " << it.first << " :\n"
+                << "F = " << -it.second << "\nM = "<<-total_torque[it.first]
+                <<"\n\n";
+    }
+  myFile.close();
+}
+
+
+template <int dim>
+void
+PWContactForce<dim>::calculate_pw_force_torque(
```

Comment:
Why recalculate the force ? Could you not get it directly from the model itself like you had done previously? Because here otherwise you will need to recode this calculation of the force on the wall for every force model.

Reply:
We gonna talk about that